core/drivers/crypto/se050/adaptors/utils/scp_config.c

```diff
@@ -0,0 +1,435 @@
+	db->ops->close(&db->fh);
+	free(db);
+
+	return res;
+}
+
+static TEE_Result scp03db_read_keys(struct se050_scp_key *keys)
+{
+	struct tee_scp03db_dir *db = calloc(1, sizeof(struct tee_scp03db_dir));
+	TEE_Result res = TEE_SUCCESS;
+	size_t len = sizeof(*keys);
+
+	if (!db)
+		return TEE_ERROR_OUT_OF_MEMORY;
+
+	db->ops = tee_svc_storage_file_ops(TEE_STORAGE_PRIVATE);
+
+	res = db->ops->open(&po, NULL, &db->fh);
+	if (res) {
+		free(db);
+		return TEE_ERROR_STORAGE_NOT_AVAILABLE;
+	}
+
+	res = db->ops->read(db->fh, 0, keys, &len);
+	if (res)
+		goto close;
+
+	if (len != sizeof(*keys))
+		res = TEE_ERROR_GENERIC;
+close:
+	db->ops->close(&db->fh);
+	free(db);
+
+	return res;
+}
+
+static sss_status_t get_id_from_ofid(uint32_t ofid, uint32_t *id)
+{
+	switch (ofid) {
+	case SE050A1_ID:
+		*id = SE050A1;
+		break;
+	case SE050A2_ID:
+		*id = SE050A2;
+		break;
+	case SE050B1_ID:
+		*id = SE050B1;
+		break;
+	case SE050B2_ID:
+		*id = SE050B2;
+		break;
+	case SE050C1_ID:
+		*id = SE050C1;
+		break;
+	case SE050C2_ID:
+		*id = SE050C2;
+		break;
+	case SE050DV_ID:
+		*id = SE050DV;
+		break;
+	default:
+		return kStatus_SSS_Fail;
+	}
+
+	return kStatus_SSS_Success;
+}
+
+static sss_status_t encrypt_key_and_get_kcv(uint8_t *enc, uint8_t *kc,
+					    uint8_t *key,
+					    struct sss_se05x_ctx *ctx,
+					    uint32_t id)
+{
+	static const uint8_t ones[] = { [0 ... AES_KEY_LEN_nBYTE - 1] = 1 };
+	uint8_t enc_len = AES_KEY_LEN_nBYTE;
+	uint8_t kc_len = AES_KEY_LEN_nBYTE;
+	sss_status_t st = kStatus_SSS_Fail;
+	sss_object_t *dek_object = NULL;
+	sss_se05x_symmetric_t symm = { };
+	sss_se05x_object_t ko = { };
+	uint8_t dek[AES_KEY_LEN_nBYTE] = { 0 };
+	size_t dek_len = sizeof(dek);
+	size_t dek_bit_len = dek_len * 8;
+
+	st = sss_se05x_key_object_init(&ko, &ctx->ks);
+	if (st != kStatus_SSS_Success)
+		return kStatus_SSS_Fail;
+
+	st = sss_se05x_key_object_allocate_handle(&ko, id,
+						  kSSS_KeyPart_Default,
+						  kSSS_CipherType_AES,
+						  AES_KEY_LEN_nBYTE,
+						  kKeyObject_Mode_Transient);
+	if (st != kStatus_SSS_Success)
+		return kStatus_SSS_Fail;
+
+	st = sss_se05x_key_store_set_key(&ctx->ks, &ko, key, AES_KEY_LEN_nBYTE,
+					 AES_KEY_LEN_nBYTE * 8, NULL, 0);
+	if (st != kStatus_SSS_Success)
+		goto error;
+
+	st = sss_se05x_symmetric_context_init(&symm, &ctx->session, &ko,
+					      kAlgorithm_SSS_AES_ECB,
+					      kMode_SSS_Encrypt);
+	if (st != kStatus_SSS_Success)
+		goto error;
+
+	st = sss_se05x_cipher_one_go(&symm, NULL, 0, ones, kc, kc_len);
+	if (st != kStatus_SSS_Success)
+		goto error;
+
+	dek_object = &ctx->open_ctx.auth.ctx.scp03.pStatic_ctx->Dek;
+	if (se050_host_key_store_get_key(&ctx->host_ks, dek_object,
+					 dek, &dek_len, &dek_bit_len))
+		goto error;
+
+	st = sss_se05x_key_store_set_key(&ctx->ks, &ko, dek, AES_KEY_LEN_nBYTE,
+					 AES_KEY_LEN_nBYTE * 8, NULL, 0);
+	if (st != kStatus_SSS_Success)
+		goto error;
+
+	st = sss_se05x_cipher_one_go(&symm, NULL, 0, key, enc, enc_len);
```

Comment:
should be rename from `error` to 'out`. This is not a error only path.

Reply:
ok